### PR TITLE
Delfin design doc update

### PR DESCRIPTION
**Pull request by NajmudheenCT**
**What type of PR is this?**

> /kind design

**What this PR does / why we need it**:
1. Made this doc as the overall delfin design doc
2. Linked module architecture design doc to this doc
3. Linked other analysis docs to this doc.
4. Renamed document name
5. Removed some duplicate information’s which are present in other module design docs
6. added use case context section
7. added resource query workflow (sequence)

**Which issue(s) this PR fixes**:
Na

**Test Report Added?**:
NA

**Test Report**:
NA
**Special notes for your reviewer**:

### CODE REVIEW

File: arch-design/delfin/RequirementsDesignSpec.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
+﻿
+# Delfin (SODA Infrastructure Manager) Requirements & Design
+**Authors:** [Najmudheen CT](https://github.com/NajmudheenCT), [Sanil Kumar D](https://github.com/skdwriting), [Xulin](https://github.com/wisererik),  [Ashit Kumar](https://github.com/kumarashit)
+
+This documentation serves as the design spec for SODA Infrastructure Management project Delfin .Delfin is a basic solution for managing heterogeneous storage systems. It uses the standard API model to centrally manage heterogeneous storage systems from different vendors and can be used as a basic framework for other SRM platforms to collect and manage storage metadata. In addition, open southbound standards are provided to help developers quickly develop and access storage product
```
Comment (AmitRoushan) [inline arch-design/delfin/RequirementsDesignSpec.md:5]: nit: full stop is misplaced

File: arch-design/delfin/RequirementsDesignSpec.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+Alert Manager module architecture [here](https://github.com/sodafoundation/architecture-analysis/blob/8549aa6ca5bf7296dedbba7b62ef63e0550a83f2/specs/SIM/alert_manager/SODA_AlertManagerDesign.md) 
+
+Driver Manager module architecture [here](https://github.com/sodafoundation/architecture-analysis/blob/694d4aa31dd4577aca5a67668b96f615c4f34be9/specs/SIM/DriverManager/DriverManager.md) 
+
+
+## Detailed Design
+
+### Use case View
+
+![Usecase](resources/Delfin_usecase.jpg)
```
Comment (AmitRoushan) [inline arch-design/delfin/RequirementsDesignSpec.md:143]: IMO, you have maintained Delfin_usecase.jpg file in "Resources" directory. Broken link issue.

File: arch-design/delfin/RequirementsDesignSpec.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+### Storage registration
+![Register](resources/Register.png)
+
+### Resource synchronization 
+
+![Resource sync](resources/Sync.jpg)
+
+### Query resource information 
+
+![Resource sync](resources/Query_workflow.png)
```
Comment (AmitRoushan) [inline arch-design/delfin/RequirementsDesignSpec.md:253]: IMO, Its a broken link as Query_workflow.jpg file is available in "Resources" directory.

File: arch-design/delfin/RequirementsDesignSpec.md [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+### Input Requirements
+1. Storage infrastructure devices and resources monitoring
+2. Performance metrics monitoring
+3. Alert reporting
+
+### Feature Requirements
+**Storage Device Monitoring** 
+
+Registering a storage device for monitoring
```
Comment (AmitRoushan) [inline arch-design/delfin/RequirementsDesignSpec.md:32]: nit: IMO, These pointers can go as bullet points.

File: arch-design/delfin/RequirementsDesignSpec.md [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+* Register alert source to receive alerts
+* Report alerts
+* Clear alert
+
+
+##### Non Functional Requirements
+**Performance requirements**
+
+NA
+**Security requirements**
```
Comment (AmitRoushan) [inline arch-design/delfin/RequirementsDesignSpec.md:98]: I believe there should be a line gap between NA and "**Security requirement**"

File: arch-design/delfin/RequirementsDesignSpec.md [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+**Other non functional requirements (scalability  , HA etc…)**
+NA
+
+
+## Architecture Analysis
+### System Architecture
+![Architecture](./resources/Architecture.jpg)
+
+### Module Architecture
+NA
```
Comment (AmitRoushan) [inline arch-design/delfin/RequirementsDesignSpec.md:111]: I believe there should be a line gap between NA and "Module Architecture"

File: arch-design/delfin/RequirementsDesignSpec.md [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+
+## Detailed Design
+
+### Use case View
+
+![Usecase](resources/Delfin_usecase.jpg)
+
+#### List of Typical Use-cases
+
+1. **Adding Storage Devices for monitoring**:- 	User call APIS by specifying  device access information such as EMC VMax, Hitachi Virtual Storage Platform, or HP 3PAR etc...
```
Comment (AmitRoushan) [inline arch-design/delfin/RequirementsDesignSpec.md:147]: alignment is inconsistent with rest of the point and ":-" can be removed

File: arch-design/delfin/RequirementsDesignSpec.md [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+
+### Interface Model
+
+
+#### External Interfaces
+REST API is provided for configuring devices for monitoring and for querying collected information.  
+Detailed API spec is available here [here](https://github.com/sodafoundation/delfin/blob/master/openapi-spec/swagger.yaml) 
+
+#### Internal Interfaces
+* Driver Interface : Provides class interfaces for the drivers to implement back end attributes collection. [Driver APIs](https://github.com/sodafoundation/delfin/blob/master/delfin/drivers/api.py)
```
Comment (AmitRoushan) [inline arch-design/delfin/RequirementsDesignSpec.md:204]: IMHO, Driver interface should be "https://github.com/sodafoundation/delfin/blob/master/delfin/drivers/driver.py"

**Review by AmitRoushan — CHANGES_REQUESTED** [review]

**Comment by AmitRoushan** [comment]
IMHO, two resources folder can be merged in one to avoid confusion.

File: arch-design/delfin/RequirementsDesignSpec.md (hunk 1 above)
Comment (NajmudheenCT) [reply arch-design/delfin/RequirementsDesignSpec.md]: Done

File: arch-design/delfin/RequirementsDesignSpec.md (hunk 2 above)
Comment (NajmudheenCT) [reply arch-design/delfin/RequirementsDesignSpec.md]: Done, thanks

File: arch-design/delfin/RequirementsDesignSpec.md (hunk 3 above)
Comment (NajmudheenCT) [reply arch-design/delfin/RequirementsDesignSpec.md]: Done , thanks

File: arch-design/delfin/RequirementsDesignSpec.md (hunk 4 above)
Comment (NajmudheenCT) [reply arch-design/delfin/RequirementsDesignSpec.md]: done

File: arch-design/delfin/RequirementsDesignSpec.md (hunk 5 above)
Comment (NajmudheenCT) [reply arch-design/delfin/RequirementsDesignSpec.md:99]: Done

File: arch-design/delfin/RequirementsDesignSpec.md (hunk 6 above)
Comment (NajmudheenCT) [reply arch-design/delfin/RequirementsDesignSpec.md:113]: Done

File: arch-design/delfin/RequirementsDesignSpec.md (hunk 7 above)
Comment (NajmudheenCT) [reply arch-design/delfin/RequirementsDesignSpec.md]: Done

File: arch-design/delfin/RequirementsDesignSpec.md (hunk 8 above)
Comment (NajmudheenCT) [reply arch-design/delfin/RequirementsDesignSpec.md]: Thanks, changed

**Review by AmitRoushan — APPROVED** [review]
LGTM

File: arch-design/delfin/RequirementsDesignSpec.md [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+
+## Architecture Analysis
+### System Architecture
+![Architecture](./resources/Architecture.jpg)
+
+### Module Architecture
+
+NA
+
+### Architecture Tenets
```
Comment (kumarashit) [inline arch-design/delfin/RequirementsDesignSpec.md:115]: Please make it bullet points. Just for better appearance

Comment (NajmudheenCT) [reply arch-design/delfin/RequirementsDesignSpec.md:115]: Done

**Review by skdwriting — APPROVED** [review]
LGTM

**Review by skdwriting — APPROVED** [review]
LGTM

**Review by kumarashit — APPROVED** [review]
LGTM

**Pull request merged**